doc/api/assert.markdown

```diff
@@ -2,42 +2,130 @@
 
     Stability: 3 - Locked
 
-This module is used so that Node.js can test itself. It can be accessed with
-`require('assert')`. However, it is recommended that a userland assertion
-library be used instead.
+The `assert` module is for use with Node.js' own unit tests. While it can be
+used by JavaScript running within Node.js by using `require('assert')`, it is
+recommended that a third-party ("userland") assertion module be used instead.
```

Comment:
`third-party` may be problematic. If the user writes their own userland assertion module, then it's second-party, right? I may be overthinking this.